compiler/modules/nback/scaffold.bal

```diff
@@ -106,8 +106,13 @@ type ModuleDI record {|
     DIFile[] files;
     DICompileUnit compileUnit;
     DISubroutineType funcType;
+    boolean debugFull;
 |};
 
+type DebugLocationConfiguration record {|
+    "fileOnly"? location = ();
+    "debugFull"? ifDebugLevel = ();
+|};
```

Comment:
I haven't made myself clear. It never makes sense to say both

- this is only when debugFull is true, and
- only include file information

So there are only three possibilities.  We need *one* argument that can take three possible values, not two arguments that can each take two values.

The question is what to call this one argument and its values.

Reply:
The way to this cleanly is to make the argument to setDebugLocation describe what the debug information is being provided for, which is one of

* error constructor - filename for backtrace (because position has to come from the error)
* call instruction - backtrace
* other instruction - debug

Better to use integers for these, so you can test for multiple ones using a comparison.

You need to think above what happens for the first case, when we have debugFull.

Reply:
fixed